src/group_theory/group_ring.lean

```diff
@@ -0,0 +1,215 @@
+/-
+Copyright (c) 2021 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+
+import algebra.monoid_algebra.basic
+import tactic.basic
+
+/-!
+# Group rings
+
+This file defines the group ring `k[G]` of a group `G` with coefficients in a ring `k`. A
+`G`-module with a compatible `k`-module structure is an `R[G]`-module.
+
+We develop an API allowing us to show `k[Gⁿ]` is a free `k[G]`-module for `n ≥ 1`. This
+will be used to construct a projective resolution of the trivial `k[G]`-module `k`.
+
```

Comment:
I have tried to do this but felt a bit silly as there's not much interesting in the file yet. Maybe after future PRs I could replace the things I selected with more important definitions.

Reply:
I guess the plan is to add to this file later on? I think it would be fine to write a "weak" set of "main definitions" now and just add to the list later as the file gets bigger.